### PR TITLE
New theme fixes

### LINKED ISSUE #4407: [PRE-RELEASE] Light & Dark Theme - Contrast issues
## Requirements
All text elements, images containing text and relevant elements without text (e.g. icons) a **minimum colour contrast ratio** of **4.5:1** is required (to fulfil **level AA** of the WCAG). For content with a font size of at least 18 Point or 14 Point plus bold a colour contrast ratio of 3.0:1 is required. 
**Disabled elements** (like disabled toolbar icons or disabled edit fields) are **excluded** from measurement.

**International standards:**
- ISO 9241-171 - 10.4.5
- WCAG 2.0 - 1.4.3

I checked the colour contrast in KeePassXC and for most elements the colour contrast is fulfilling the requirements. So here I list only those elements where the colour contrast is not sufficient.
I know there is a gap between design (what it looking good and sexy) and the required colour contrast. But these requirements are important for all users, not only for users that have visual impairments.

## Results for the Light Theme
In the **Light Theme** the following elements do **not fulfil the requirements** described above (the colour contrast ratio is less than 4.5:1 at these elements):

1. Selected item in the menu bar (like "Database") and in the context menu: **3.5:1**
2. Tooltip on toolbar-icons: **3.8:1**
3. Green Button "OK": **3.3:1**
4. Green headlines of group boxes in Search-Help: **3.5:1**
5. Green headlines of group boxes when editing an entry (like "Additional Attributes"): **3.4:1**
6. Arrows to expand/collapse the groups: **4.2:1** (maybe you take the same colour (#686970) that you are using for the arrows that indicate the sort-order in tables, they have 4.7:1)
7. URLs in Entry-Preview-Panel: **3.2:1**
8. Selected entry in lists (like groups, list of entries) wen the list **is focussed** (I clicked on a group and measured the selected group): **3.5:1**
9. Selected entry in lists (like groups, list of entries) wen the list **is not focussed** (I clicked on a group and measured the selected entry in the password list): **3.6:1**
10. Arrow-Icon in the combo box "Username": **3.7:1**
11. *Password Quality "Weak" in the "Password Generator" tool: **2.4:1***
12. *Password Quality "Good" in the "Password Generator" tool: **3.2:1***
13. *Password Quality "Excellent" in the "Password Generator" tool: **4.2:1***
14. *Password Quality "Weak" when editing a password: **2.2:1***
15. *Password Quality "Good" when editing a password: **3.0:1***
16. *Password Quality "Excellent" when editing a password: **3.9:1***
17. Links in the settings (Browser integration) and in the About window: **3.2:1**

## Results for the Dark Theme
In the **Dark Theme** the following elements do **not fulfil the requirements** described above:

18. Arrows to expand/collapse the groups: **3.3:1**
19. Arrows that indicate the sort-order in tables: **2.7:1**
20. URLs in Entry-Preview-Panel: **4.2:1**
21. Arrow-Icon in the combo box "Username": **3.6:1**
22. *Password Quality "Poor" in the "Password Generator" tool: **2.6:1***
23. *Password Quality "Good" in the "Password Generator" tool: **4.2:1***
24. *Password Quality "Excellent" in the "Password Generator" tool: **3.2:1***
25. *Password Quality "Poor" when editing a password: **2.2:1***
26. *Password Quality "Good" when editing a password: **3.5:1****
27. *Password Quality "Excellent" when editing a password: **2.6:1***
28. Links in the settings (Browser integration) and in the About window: **4.2:1**

**Annotations**
* At the numbers 11 to 16 (in the light theme) and 22 to 27 (dark theme) I measured the contrast of the coloured bar to the background (below the bar).

* I would say that the **password quality bar** is an **decorative element** (because users also see the text) so in my opinion there is nothing that should be changed here.
This is the reason, why I wrote these entries in *cursive text*

## Debug Info
KeePassXC - Version 2.5.3-snapshot
Build Type: Snapshot
Revision: 557736e

Qt 5.12.0
Debugging mode is disabled.

Operating system: Windows 10 (10.0)
CPU architecture: x86_64
Kernel: winnt 10.0.18363

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
 libgcrypt 1.8.4

### LINKED ISSUE #4637: Focus indication is insufficient
## Overview
When both Group and Account are focused, either by mouse or by keyboard, it is impossible to discern, which panel has focus, as both entries (Group and Account) are highlighted by selected item colour (within tree/listbox). More over it's impossible to discern if either panel has focus at all.

## Steps to Reproduce
  * Click onto any Group in Group panel, background becomes highlighted (by default blue).
  * Click onto any Account in Account panel, background becomes highlighted (by default blue).
  * Click into searchbox.
  * Step away from computer and turn around once in a room.
  * Sit back and look at KeepassXC.
  * Reply quickly (under 1 second) just by looking at GUI, which element is focused?

## Expected Behavior
Focused elements in unfocused panels should change highlight color to indicate selection, but not focus, eg. by default they should have dark gray background or something similar.

Each panel could also have small header line above it's rectangle or some other way to indicate it has focus (dark blue (focused) or gray (unfocused)).

## Actual Behavior
Nothing. All selected elements appear focused. It is impossible to tell quickly what has focus.

## Context
Daily use.

KeePassXC - 2.5.3
Revision: f8c962b

Operating System: Linux
Desktop Env: none/WM
Windowing System: X11

### PR DESCRIPTION
**Pull request by phoerious**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )

- Increases colour contrast to match WCAG AA requirements. Fixes #4407.
- Improve contrast between active and inactive highlight. Fixes #4637
- Fixes entry reference colour being incompatible with dark mode.

The green button in the light theme has become a bit heavy due to the required lightness changes, but I tried to keep the visual changes to a minimum.

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )

image

image

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
I think we need a common way to obtain colors for the following "states": Error, Warning, Information, Positive

Some additional issues:

1. Error color on URL edit field:
image

2. Colors on database reports in dark mode:
image

3. Search bar can be hidden (need to increase min-size of window) and the overflow arrows are hard to see:
image

**Comment by phoerious** [comment]
I was thinking about an extended palette with those colours.

**Comment by droidmonkey** [comment]
New database wizard needs to be fixed (somehow) in dark mode:

image

**Comment by phoerious** [comment]
image

I fixed the QWizard issue by overriding the QFrame palette with colours derived from the default palette. Qt itself sets a hard-coded `QColor(255, 255, 255, 153)`, which breaks not only our dark theme, but also native dark modes on macOS and Linux.

There are various Qt bug reports about this, but they all reference each other, some are closed, some are not, it's a mess. From what I understand, it's supposed to be fixed in 5.12, but that is obviously not the case. The two most relevant reports are probably https://bugreports.qt.io/browse/QTBUG-70346 and https://codereview.qt-project.org/c/qt/qtbase/+/239007.

File: src/gui/styles/base/basestyle.qss [hunk 1]
```diff
@@ -1,8 +1,13 @@
-QPushButton:default {
+QPushButton:default, QDialog QPushButton:default {
```
Comment (phoerious) [inline src/gui/styles/base/basestyle.qss]: Shouldn't this be redundant?

Comment (droidmonkey) [reply src/gui/styles/base/basestyle.qss]: Oops that was for testing, will remove.

**Comment by phoerious** [comment]
Branch should be ready now.

State colours are now managed by a palette and the health colours in particular are a little easier on the eyes now.

image

**Comment by droidmonkey** [comment]
Found an awkward anomaly on macOS, not sure if we can fix this?

image

**Comment by phoerious** [comment]
I don't think so.

**Comment by droidmonkey** [comment]
Here are comparisons of the new highlight colors:

image

image

**Comment by phoerious** [comment]
Amended a few more changes, we should definitely merge this soon before other branches start diverging too much.

**Comment by droidmonkey** [comment]
Looked through everything an tested on windows. Looks fantastic.

**Comment by wombat01** [comment]
Couldn't a dark header be made at the header, please?

image

Some as:
image

**Comment by phoerious** [comment]
We tried it. It's not as easy as you might think.

**Pull request merged**